### ISSUE TITLE
Userland pixel shader language

### ISSUE THREAD

**Issue by osnr** [FolkComputer/folk#106]
so you can shade a page (or region) with a one-liner:

<img width="504" alt="image" src="https://github.com/FolkComputer/folk/assets/96857/f5bce082-b3c0-4303-8574-f9f770832c1c">

**Comment by cwervo** [comment]
@osnr thanks to your push on Vulkan, should we close this?

**Comment by cwervo** [comment]
Or, was your point that we could add sugar to make the shader interface cleaner?

**Comment by osnr** [comment]
> Or, was your point that we could add sugar to make the shader interface cleaner?

Yeah. This issue is _because_ of Vulkan, the shader interface is still too tedious to use for simple toys

**Comment by cwervo** [comment]
Adding onto this:

- [ ] Should auto include functions claimed before the userland shader `Wish`
- [ ] Should add a default set of useful uniforms (at least `u_time` and `u_resolution`)

There's a bunch of other defaults I'm thinking of, will sketch this out more next week when I have more time
